### DISCUSSION TITLE
Is there a way to disable JVM-like stack traces?

### DISCUSSION THREAD

**Discussion by nreid260** [oracle/graal#10249] [Q&A]
We're transforming the Kotlin compiler into a native-image. This works, but the resulting ELF is quite large. From decompiling some of the functions, we see that there are a lot of instructions for handling `StackOverflowException`s copied into every function. We also observe that during crashes, we get stack trace listing JVM classnames and line numbers.

This begs the question: Can we get native stacktraces instead? Seemingly this would seemingly unblock two significant improvements:
  - A lot of instructions for `StackOverflowException`s would be obviated
  - The classname strings could be removed from the binary

I might be missing something here, but these are tradeoffs I'd gladly make.

**Comment by alina-yur** [comment]
Hi @nreid260, to answer both your questions, there are two things I would suggest:
— You can use [Build Reports](https://www.graalvm.org/latest/reference-manual/native-image/overview/build-report/) to see which parts of the application contribute the most to the image size, for code area you can get breakdown by bytecode size and by method count.
— Another things to try would be using the `-Os` flag ("optimize for size"), it can reduce the image size for up to ~30%. 

Unfortunately, at the moment there is no option to disable stack traces. If you have a specific case where the stack overflow increases the size significantly, it would be helpful if you could share with us a reproducer.

**Reply by nreid260** [reply]
When I try passing `-Os`, I get an error back: `Error: Invalid option '-Os'. Accepted values are 'b' or numeric value >= 0.`

`build-report` is not available in the Community edition. However, I was able to use standard debugging tools to inspect the function and heap sizes. The heap-dump overview can be seen https://github.com/oracle/graal/discussions/10248 The function sizes appear not terrible.

kotlinc is pretty bloaty, so it's hard to tell which parts are actually needed. There are 140K functions to sift though.

**Comment by alina-yur** [comment]
>When I try passing -Os, I get an error back: Error: Invalid option '-Os'. Accepted values are 'b' or numeric value >= 0.

You must be on a version <23. In that case you can either upgrade, or if not possible use `upx` for compression.
